File: cses/permutations.py
# CSES - Permutations
# https://cses.fi/problemset/task/1070

if __name__ == "__main__":
	n = int(input())
	if n == 2 or n == 3:
		print("NO SOLUTION")
	else:
		if n % 2 == 0:
			even = n
			odd = n - 1
		else:
			odd = n
			even = n - 1
		for i in range(odd, 0, -2):
			print(i, '', end='')
		for i in range(even, 1, -2):
			print(i, '', end='')
		print()

	

File: cses/permutations.cpp
// CSES - Permutations
// https://cses.fi/problemset/task/1070

#include <iostream>

using namespace std;

int main()
{
	int n;

	cin >> n;
	if (n == 2 || n == 3)	{
		cout << "NO SOLUTION\n";
	}
	else	{
		int even, odd;

		if (n % 2 == 0)	{
			even = n;
			odd = n - 1;
		}
		else	{
			odd = n;
			even = n - 1;
		}
		for (int i = odd; i > 0; i -= 2)	{
			cout << i << " ";
		}
		for (int i = even; i > 1; i -= 2)	{
			cout << i << " ";
		}
		cout << "\n";
	}
	return 0;
}

